src/IonParserBinaryRaw.ts

```diff
@@ -208,74 +211,26 @@ export class ParserBinaryRaw {
         return ParserBinaryRaw._readVarSignedIntFrom(this._in);
     }
 
-    private readVarLongInt() : LongInt {
-        let bytes = [];
-        let byte = this._in.next();
-        let isNegative = (byte & 0x40) !== 0;
-        let stopBit = byte & 0x80;
-        bytes.push(byte & 0x3F);  // clears the sign/stop bit
-        while(!stopBit) {
-            byte = this._in.next();
-            stopBit = byte & 0x80;
-            byte &= 0x7F;
-            bytes.push(byte);
-        }
-        return LongInt.fromVarIntBytes(bytes, isNegative);
-    }
-
-    static _readSignedIntFrom(input: BinarySpan, numberOfBytes: number) : LongInt {
+    static _readSignedIntFrom(input: BinarySpan, numberOfBytes: number) : SignAndMagnitudeInt {
         if (numberOfBytes == 0) {
-            return new LongInt(0);
+            return new SignAndMagnitudeInt(JsbiSupport.ZERO);
         }
         let bytes: Uint8Array = input.view(numberOfBytes);
         let isNegative = (bytes[0] & 0x80) == 0x80;
         let numbers = Array.prototype.slice.call(bytes);
         numbers[0] = bytes[0] & 0x7F;
-        return LongInt.fromIntBytes(numbers, isNegative);
-    }
-
-    private readSignedInt() : LongInt {
-        return ParserBinaryRaw._readSignedIntFrom(this._in, this._len);
+        let magnitude: JSBI = JsbiSerde.fromUnsignedBytes(numbers);
+        return new SignAndMagnitudeInt(magnitude, isNegative);
     }
 
-    static _readUnsignedIntFrom(input: BinarySpan, numberOfBytes: number) : number {
-        let value = 0, bytesRead = 0, byte;
-        if (numberOfBytes < 1)
-            return 0;
-
-        while (bytesRead < numberOfBytes) {
-            byte = input.next();
-            bytesRead++;
-            value = value << 8;
-            value = value | (byte & 0xff);
-        }
-
-        // We overflowed
-        if (numberOfBytes > 4 || value < 0) {
-            throw new Error("Attempted to read an unsigned int that was larger than 31 bits."
-                + " Use readUnsignedLongIntFrom instead. UInt size: " + numberOfBytes + ", value: " + value
-            );
-        }
-        // Fewer bytes than the required `numberOfBytes` were available in the input
-        if (byte === EOF) {
-            throw new Error("Ran out of data while reading a " + numberOfBytes + "-byte unsigned int.");
-        }
-
-        return value;
+    static _readUnsignedIntFrom(input: BinarySpan, numberOfBytes: number) : JSBI {
```

Comment:
We should not be replacing the lower level ints with JSBI we should only be creating JSBI values for decimal/timestamp decimals and Ion INTs lower level values should still be consumed as numbers.

Reply:
As discussed offline, the `UInt` and `Int` primitives are only used [in the `int`, `decimal`, and `symbol` types](http://amzn.github.io/ion-docs/docs/binary.html#uint-and-int-fields) (and `timestamp`, which has a `decimal` field), so this isn't as far-reaching a change as it seemed at first glance.